exporter/observiqexporter/converter.go

```diff
@@ -39,10 +51,193 @@ type observIQLogEntry struct {
 }
 
+// Hash related variables, re-used to avoid multiple allocations
+var fnvHash = fnv.New128a()
+var fnvHashOut = make([]byte, 0, 16)
+
 // Convert pdata.Logs to observIQLogBatch
-func logdataToObservIQFormat(ld pdata.Logs, agentID string, agentName string) (*observIQLogBatch, []error) {
-	return &observIQLogBatch{}, []error{}
+func logdataToObservIQFormat(ld pdata.Logs, agentID string, agentName string, buildVersion string) (*observIQLogBatch, []error) {
+	var rls = ld.ResourceLogs()
+	var sliceOut = make([]*observIQLog, 0, ld.LogRecordCount())
+	var errorsOut = make([]error, 0)
+
+	for i := 0; i < rls.Len(); i++ {
+		rl := rls.At(i)
+		res := rl.Resource()
+		resMap := attributeMapToBaseType(res.Attributes())
+		ills := rl.InstrumentationLibraryLogs()
+		for j := 0; j < ills.Len(); j++ {
+			ill := ills.At(j)
+			logs := ill.Logs()
+			for k := 0; k < logs.Len(); k++ {
+				oiqLogEntry := resourceAndInstrumentationLogToEntry(resMap, logs.At(k), agentID, agentName, buildVersion)
+
+				jsonOIQLogEntry, err := json.Marshal(oiqLogEntry)
+
+				if err != nil {
+					//Skip this log, keep record of error
+					errorsOut = append(errorsOut, consumererror.Permanent(err))
+					continue
+				}
+
+				//fnv sum of the message is ID
+				fnvHash.Reset()
+				_, err = fnvHash.Write(jsonOIQLogEntry)
+				if err != nil {
+					errorsOut = append(errorsOut, consumererror.Permanent(err))
+					continue
+				}
+
+				fnvHashOut = fnvHashOut[:0]
+				fnvHashOut = fnvHash.Sum(fnvHashOut)
+
+				fnvHashAsHex := hex.EncodeToString(fnvHashOut)
+
+				sliceOut = append(sliceOut, &observIQLog{
+					ID:    fnvHashAsHex,
+					Size:  len(jsonOIQLogEntry),
+					Entry: preEncodedJSON(jsonOIQLogEntry),
+				})
+			}
+		}
+	}
+
+	return &observIQLogBatch{Logs: sliceOut}, errorsOut
+}
+
+// Output timestamp format, an ISO8601 compliant timestamp with millisecond precision
+const timestampFieldOutputLayout = "2006-01-02T15:04:05.000Z07:00"
+
+func resourceAndInstrumentationLogToEntry(resMap interface{}, log pdata.LogRecord, agentID string, agentName string, buildVersion string) *observIQLogEntry {
+	return &observIQLogEntry{
+		Timestamp: timestampFromRecord(log),
+		Severity:  severityFromRecord(log),
+		Resource:  resMap,
+		Message:   messageFromRecord(log),
+		Data:      attributeMapToBaseType(log.Attributes()),
+		Body:      bodyFromRecord(log),
+		Agent:     &observIQAgentInfo{Name: agentName, ID: agentID, Version: buildVersion},
+	}
+}
+
+func timestampFromRecord(log pdata.LogRecord) string {
+	if log.Timestamp() == 0 {
+		return timeNow().UTC().Format(timestampFieldOutputLayout)
+	}
+	return log.Timestamp().AsTime().UTC().Format(timestampFieldOutputLayout)
+}
+
+func messageFromRecord(log pdata.LogRecord) string {
+	if log.Body().Type() == pdata.AttributeValueTypeString {
+		return log.Body().StringVal()
+	}
+
+	return ""
+}
+
+// If Body is not a string, it is suitable to be used on the observIQ log entry as "body"
```

Comment:
Is this supposed to read?
```suggestion
// If Body is not a string, it is not suitable to be used on the observIQ log entry as "body"
```